packages/grid/_modules_/grid/hooks/root/useGridApiContext.ts

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { GridApiContext } from '../../components/GridApiContext';
+
+export function useGridApiContext() {
+  const apiRef = React.useContext(GridApiContext);
+
+  if (apiRef === undefined) {
```

Comment:
What do you think about raising a warning instead of throwing and only in dev mode to not have any bundle size impact? Is it important enough to throw?

Reply:
I think it is important to throw as users would have issues at a later stage.

We could also consider combining it with useGridApiRef so you have a single point to get or create the apiRef 🤔

Reply:
> What do you think about raising a warning instead of throwing and only in dev mode to not have any bundle size impact?

@oliviertassinari  I tested with the docs and the warning wasn't sent to the console, but to the terminal that was running the dev server. A throw in this situation facilitates for the user to know where's the problem.

> We could also consider combining it with useGridApiRef so you have a single point to get or create the apiRef 🤔

@dtassone Let's imagine that there's another grid inside the grid, like a child row. How will we know if a apiRef has to be created or the one in the context should used? Maybe we need to add an argument to force to create the apiRef even if there's a context in the tree.

Reply:
> but to the terminal that was running the dev server.

@m4theushw True, I was actually just talking about this problem in https://github.com/mui-org/material-ui/issues/18119#issuecomment-860877215. It's a Next.js thing only, that they could solve by forwarding the console log too (not only the error).

> How will we know if a apiRef has to be created or the one in the context should used?

💯. I would even push it one step further is unbundle useGridApiRef. I think that we need 3 hooks not 1.

- `useGridApiContext` to consume the API ref
- `const apiRef = React.useRef<GridApi>(createGridApi());` to create the apiRef and store it.
- `useGridApiRef`to read the API ref from the child data grid **ONLY**

I did a first step in this direction with #990